lib/coinbase/wallet.rb

```diff
@@ -49,26 +51,39 @@ def wallets_api
     # User#import_wallet.
     # @param model [Coinbase::Client::Wallet] The underlying Wallet object
     # @param seed [String] (Optional) The seed to use for the Wallet. Expects a 32-byte hexadecimal with no 0x prefix.
-    #   If not provided, a new seed will be generated.
-    # @param address_count [Integer] (Optional) The number of addresses already registered for the Wallet.
+    #   If nil, a new seed will be generated. If the empty string, no seed is generated, and the Wallet will be
+    #   unhydrated.
+    # @param address_models [Array<Coinbase::Client::Address>] (Optional) The models of the addresses already registered
+    #   with the Wallet. If not provided, the Wallet will derive the first default address.
     # @param client [Jimson::Client] (Optional) The JSON RPC client to use for interacting with the Network
-    def initialize(model, seed: nil, address_count: 0)
-      raise ArgumentError, 'Seed must be 32 bytes' if !seed.nil? && seed.length != 64
+    def initialize(model, seed: nil, address_models: [])
+      raise ArgumentError, 'Seed must be 32 bytes' if !seed.nil? && !seed.empty? && seed.length != 64
+      raise ArgumentError, 'Seed must be present if address_models are provided' if seed.nil? && address_models.any?
+      raise ArgumentError, 'Seed must be empty if address_models are not provided' if !seed.nil? &&
+                                                                                      seed.empty? &&
+                                                                                      address_models.empty?
```

Comment:
Guard clauses with multiple boolean expressions like this can be difficult to read.

If we can't get a sufficiently succinct guard clause that fits on one line, would prefer something more readable like:

```
      raise ArgumentError, 'Seed must be 32 bytes' if valid_seed_size?(seed)

      if seed_and_addresses_omitted?(seed, address_models)
        raise ArgumentError, 'Seed must be present if address_models are provided'
      end

      if seed_and_addresses_provided?(seed, address_models)
        raise ArgumentError, 'Seed must be empty if address_models are not provided'
      end
```

Can replace the `_omitted` and `_provided` with the actual expressions if preferred too, but any wrapping of the guard clause like that is painful to read

Reply:
Tidied this up.